### ISSUE TITLE
Modernize random number generation (i.e., stop using np.random.seed/get_state/set_tate)

### ISSUE THREAD

**Issue by eteq** [astropy/astropy#11687]
`numpy` has gone away from the `np.random.seed`/`np.random.set_state`/`np.random.get_state` models for random number generation.  `astropy` should follow suit where possible.  This in practice means the following tasks:

- [ ] Stop using our home-grown `RNGContext` in the tests and replace with the newer `numpy` mechanism.
- [ ] Replace anywhere that `np.random.*` is used currently with something that uses the new mechanism (e.g. the random functions in #11628 ), which may require some API additions to let the new numpy generators be passed into functions.
- [ ] ? Deprecate our home-grown `RNGContext` and replacing it with the "new" numpy mechanism.  Note it is *not* currently clear to me if we can do this.  I don't think the numpy scheme supports a context manager a la our `ScienceState` approach. In that case we have to decide if we think it's worth supporting this long-term, trying to get this into `numpy` or just deciding it's better to use the `numpy` approach anyway.

For some backstory/justification on the `numpy` shift see:
* https://numpy.org/doc/stable/reference/random/index.html - the paragraph beginning with "Since Numpy version 1.17.0"
* https://numpy.org/doc/stable/reference/random/new-or-different.html#new-or-different
* https://towardsdatascience.com/stop-using-numpy-random-seed-581a9972805f

cc @adrn @lpsinger @mhvk (as discussers from #11628)

**Comment by pllim** [comment]
Might be relevant: #10644 and #11566

**Comment by mhvk** [comment]
Yes, good idea. Note that one of the specific numpy generators - `Philox` - is explicitly meant for being able to (re)start at a given position.

**Comment by Maia-jp** [comment]
Is this issue is still open?

**Comment by pllim** [comment]
Yes, a previous effort at https://github.com/astropy/astropy/pull/12669 was abandoned.

**Comment by adrn** [comment]
🧟 back from the dead!

Adding a note to say that [SPEC 7](https://scientific-python.org/specs/spec-0007/) is very relevant here.
